File: src/modules/user-management/domain/events/account-restricted.event.ts
import { BaseDomainEvent } from '@shared/domain/events/base-domain-event';

export class AccountRestrictedEvent extends BaseDomainEvent {
  public readonly eventName = 'account.restricted';
  constructor(
    public readonly userId: string,
    public readonly reason: string,
  ) { super(); }
}

File: src/main/java/com/dddrivebye/usermanagement/domain/event/AccountRestrictedEvent.java
package com.dddrivebye.usermanagement.domain.event;

import com.dddrivebye.shared.domain.event.BaseDomainEvent;
import com.dddrivebye.usermanagement.domain.valueobject.UserId;

public final class AccountRestrictedEvent extends BaseDomainEvent {

    private final UserId userId;

    public AccountRestrictedEvent(UserId userId) {
        this.userId = userId;
    }

    public UserId userId() {
        return userId;
    }
}
